OpenSim/Actuators/VandenBogert2011Muscle.cpp

```diff
@@ -0,0 +1,1234 @@
+            muscleLength, y[0], y[1], ydot_guess[0], ydot_guess[1], u,
+            returnJacobians);
+    return Results;
+
+}
+
+//------------------------------------------------------------------------------
+VandenBogert2011Muscle::ImplicitResidual VandenBogert2011Muscle::
+calcImplicitResidual(double muscleLength, double projFibLenNorm, double activ,
+                     double projFibVelNorm, double activdot, double u,
+                     int returnJacobians=0) const {
+
+
+    //TODO: Match symbols to doxygen diagram and Add symbolic equations comments
+    //TODO: May want to make this into a seperate function
+
+    // -------------------------Parameters----------------------------//
+    //F_{iso} Maximum isometric force that the fibers can generate
+    double maxIsoForce = getMaxIsometricForce();
+
+    //u_{max} (dimensionless) strain in the series elastic element at load of
+    //      maxIsometricForce
+    double fMaxTendonStrain = getFMaxTendonStrain(); //Strain in the series
+    //      elastic element at load of Fmax
+
+    //W (dimensionless) width parameter of the force-length relationship of
+    //      the muscle fiber
+    double fl_width = getFlWidth();
+
+    //AHill (dimensionless) Hill parameter of the force-velocity relationship
+    double fv_AHill= getFvAHill();   //Hill parameter of the f-v relationship
+
+    //FV_{max} (dimensionless) maximal eccentric force
+    double fv_maxMultiplier = getFvmaxMultiplier();
+    //Maximal eccentric force multiplier
+
+    // L_{opt}  (m) Optimal Length of Contractile Element;
+    double optFiberLength = getOptimalFiberLength();
+    //Optimal Length of Contractile Element
+
+    //b (s/m) damping coefficient of damper parallel to the fiber
+    // (normalized to maxIsometricForce)
+    double dampingCoeff = getDampingCoefficient();
+
+    //L_{slack,fiber} Slack length of the parallel elastic element, divided by
+    //          Lceopt
+    double fiberSlackLengthNorm= getNormFiberSlackLength();
+
+    //L_{slack,tendon} (m) slack length of the tendon
+    double tendonSlackLength = getTendonSlackLength();
+
+    //T_{act} (s) Activation time
+    double activTimeConstant = getActivTimeConstant();
+
+    //T_{deact} (s) Deactivation time
+    double deactivTimeConstant = getDeactivTimeConstant();
+
+    //phi_{opt} pennation at Optimal Fiber Length
+    double pennAtOptFiberLength = getPennAtOptFiberLength();
+
+
+    // constants derived from the muscle parameters
+    double vMaxNorm = 10 * optFiberLength;
+    //Maximum shortening velocity in optimal fiber lengths per sec
+
+    // Jacobian Matrices
+    SimTK::Mat22 df_dy;
+    SimTK::Mat22 df_dydot;
+
+
+    //-------Convert projFiberLength & Velocity to fiberLength & Velocity------/
+    double cosPenn;
+    double dcosPenn_dprojFibLen;
+    double fiberLengthNorm;
+    double dfiberLength_dprojFibLen;
+    if (pennAtOptFiberLength<0.01)  {
```

Comment:
If this threshold appears in multiple places and is not user-configurable, it could be a static const member variable.